src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -216,13 +215,45 @@ var ReactDOMInput = {
 
     var value = LinkedValueUtils.getValue(props);
     if (value != null) {
+      var node = ReactDOMComponentTree.getNodeFromInstance(inst);
+
       // Cast `value` to a string to ensure the value is set correctly. While
       // browsers typically do this as necessary, jsdom doesn't.
-      DOMPropertyOperations.setValueForProperty(
-        ReactDOMComponentTree.getNodeFromInstance(inst),
-        'value',
-        '' + value
-      );
+      var newValue = '' + value;
+
+      // To avoid side effects (such as losing text selection), only set value if changed
+      if (newValue !== node.value) {
+        node.value = newValue;
+      }
+    } else {
+      ReactDOMInput.postUpdateWrapper(inst);
+    }
+  },
+
+  postMountWrapper: function(inst) {
+    var node = ReactDOMComponentTree.getNodeFromInstance(inst);
+    node.value = node.value; // Detach value from and defaultValue
+    node.checked = node.checked; // Detach checked from and defaultChecked
+  },
+
+  postUpdateWrapper: function(inst) {
```

Comment:
Is there a reason this needs to go in a separate function rather than just in updateWrapper? (Not actually sure offhand which is preferable, but this seems to be a change.)